File: Sorting-Algorithms/Selection-Sort/selectionSort.cpp
#include <iostream>
#include <vector>

void selectionSort(std::vector<int>& array){
   for (int i = 0; i < array.size(); i++){
      int min_index = i;

      for(int j = i+1; j < array.size(); j++){
         if(array[j] < array[min_index]){
            min_index = j;
         }
      }
      int temp = array[i];
      array[i] = array[min_index];
      array[min_index] = temp;
   }
}

File: Sorting-Algorithms/Selection-Sort/selectionSort.py
"""
Selection sort is an algorithm which repeatedly selects the next-smallest element and swap it into place

Time Complexity is O(n^2)

"""

def selectionSort(array):
   for i in range(len(array)):
      min_index = i
      for j in range(i+1, len(array)):
         if array[j] < array[min_index]:
            min_index = j

      array[i], array[min_index] = array[min_index], array[i]
